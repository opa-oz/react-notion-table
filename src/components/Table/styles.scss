:local {
  .TableContainer {
    display: flex;
    flex-direction: column;
    width: 100%;
    overflow: auto;
  }

  .Table {
    display: flex;
    flex-direction: column;
  }
}
