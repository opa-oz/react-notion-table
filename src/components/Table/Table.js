import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './styles.scss';

const Table = (props) => {
	const { className = null } = props;

	return (
		<div
			className={classNames({
				[styles.Table]: true,
				[className]: !!className,
			})}
		>
			Table will be here or NOT?
		</div>
	);
};

Table.propTypes = {
	className: PropTypes.string,
};

Table.defaultProps = {
	className: null,
};

export default Table;
