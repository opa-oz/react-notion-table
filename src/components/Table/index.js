import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Cell from '../Cell';
import Row from '../Row';

import { createTable, insertRow, insertColumn } from '../../utils/actions';

import styles from './styles.scss';

const DEFAULT_SIZES = { cols: 2, rows: 2 };
const DEFAULT_TABLE = createTable(DEFAULT_SIZES);

const Table = (props) => {
	const { className } = props;
	const [sizes, setSizes] = useState(DEFAULT_SIZES);
	const [tableData, setTableData] = useState(DEFAULT_TABLE);

	const handleInsertRow = (index) => {
		setTableData(insertRow(tableData, index));
		setSizes({ ...sizes, rows: sizes.rows + 1 });
	};

	const handleInsertColumn = (index) => {
		setTableData(insertColumn(tableData, index));
		setSizes({ ...sizes, cols: sizes.cols + 1 });
	};

	const makeRow = (row) => {
		const $index = row.get('$index');
		const isHeader = row.get('isHeader');
		const cells = row.get('cells');

		return (
			<Row isHeader={isHeader} key={$index}>
				{cells.map((cell) => (
					<Cell data={cell} key={cell.get('$index')} />
				))}
			</Row>
		);
	};

	const rows = tableData.get('rows');

	return (
		<div
			className={classNames({
				[styles.TableContainer]: true,
				[className]: !!className,
			})}
		>
			<div className={styles.Table}>
				{rows.map(makeRow)}
				<button type="button" id="insertRow" onClick={() => handleInsertRow()}>
					Insert Row
				</button>
				<button
					type="button"
					id="insertColumn"
					onClick={() => handleInsertColumn()}
				>
					Insert Column
				</button>
			</div>
		</div>
	);
};

Table.propTypes = {
	className: PropTypes.string,
};

Table.defaultProps = {
	className: null,
};

export default Table;
