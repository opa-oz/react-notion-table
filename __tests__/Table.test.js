import React from 'react';
import renderer from 'react-test-renderer';

import Table from '../src/components/Table';

describe('#Table', () => {
	test('should render properly', () => {
		const component = renderer.create(<Table />);

		const table = component.toJSON();

		expect(table).toMatchSnapshot();
	});

	test('should render with additional class', () => {
		const component = renderer.create(<Table className="AdditionalClass" />);

		const table = component.toJSON();

		expect(table).toMatchSnapshot();
	});
});
