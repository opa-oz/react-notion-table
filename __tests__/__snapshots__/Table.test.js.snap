// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Table should add row to table 1`] = `
<div
  className="TableContainer AdditionalClass"
>
  <div
    className="Table"
  >
    <div
      className="Row Header"
    >
      <div
        className="Cell"
      >
        Cell
      </div>
      <div
        className="Cell"
      >
        Cell
      </div>
    </div>
    <div
      className="Row"
    >
      <div
        className="Cell"
      >
        Cell
      </div>
      <div
        className="Cell"
      >
        Cell
      </div>
    </div>
    <button
      id="insertRow"
      onClick={[Function]}
      type="button"
    >
      Insert Row
    </button>
    <button
      id="insertColumn"
      onClick={[Function]}
      type="button"
    >
      Insert Column
    </button>
  </div>
</div>
`;

exports[`#Table should render properly 1`] = `
<div
  className="TableContainer"
>
  <div
    className="Table"
  >
    <div
      className="Row Header"
    >
      <div
        className="Cell"
      >
        Cell
      </div>
      <div
        className="Cell"
      >
        Cell
      </div>
    </div>
    <div
      className="Row"
    >
      <div
        className="Cell"
      >
        Cell
      </div>
      <div
        className="Cell"
      >
        Cell
      </div>
    </div>
    <button
      id="insertRow"
      onClick={[Function]}
      type="button"
    >
      Insert Row
    </button>
    <button
      id="insertColumn"
      onClick={[Function]}
      type="button"
    >
      Insert Column
    </button>
  </div>
</div>
`;

exports[`#Table should render with additional class 1`] = `
<div
  className="TableContainer AdditionalClass"
>
  <div
    className="Table"
  >
    <div
      className="Row Header"
    >
      <div
        className="Cell"
      >
        Cell
      </div>
      <div
        className="Cell"
      >
        Cell
      </div>
    </div>
    <div
      className="Row"
    >
      <div
        className="Cell"
      >
        Cell
      </div>
      <div
        className="Cell"
      >
        Cell
      </div>
    </div>
    <button
      id="insertRow"
      onClick={[Function]}
      type="button"
    >
      Insert Row
    </button>
    <button
      id="insertColumn"
      onClick={[Function]}
      type="button"
    >
      Insert Column
    </button>
  </div>
</div>
`;
